components/LessonsChallenges.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import '../scss/lessonsChallenges.scss'
+
+type ChallengeStatusProps = {
+  challengesData: Challenge[]
+}
+
+type LessonChallengeProps = {
+  lessons: LessonChallenge[]
+}
+
+type LessonChallenge = {
+  order: number
+  title: string
+  challenges: Challenge[]
+}
+
+type Challenge = {
+  challengeNumber: number
+  challengeStatus?: string
+}
+
+export const ChallengeStatus: React.FC<ChallengeStatusProps> = ({
+  challengesData
+}) => {
+  const displayChallengeStatus = challengesData.map(
+    (eachChallenge: Challenge, challengeId: number) => {
+      let challengeStatus = 'bg-gray'
+      if (eachChallenge.challengeStatus === 'passed') {
+        challengeStatus = 'bg-success'
+      }
+      if (eachChallenge.challengeStatus === 'wrong') {
+        challengeStatus = 'bg-danger'
+      }
+      if (eachChallenge.challengeStatus === 'pending') {
+        challengeStatus = 'bg-warning'
+      }
+      return (
+        <span
+          key={challengeId}
+          className={`challenge_status ${challengeStatus}`}
+        />
+      )
+    }
+  )
+  return <div className="challenges_container">{displayChallengeStatus}</div>
+}
+
+const LessonsChallenges: React.FC<LessonChallengeProps> = ({ lessons }) => {
+  const displayLessons = lessons.map((lesson, lessonId) => {
+    const filterPassedChallenges = lesson.challenges.filter(
+      e => e.challengeStatus === 'passed'
```

Comment:
Quick tip: It's always better to name the element `e ==> lesson`

Reply:
@Wolfy64  do you mean rename e to challenge in this case?

Reply:
> @Wolfy64 do you mean rename e to challenge in this case?

What are you filter? I would say `lesson` what do you think? 
Anyway but it's always better than just `e` 🙂